### ISSUE TITLE
model.find() returning all records from a collection when passing wrong key and value 

### ISSUE THREAD

**Issue by satnamDev-Ops** [Automattic/mongoose#12006]
### Prerequisites

- [X] I have written a descriptive issue title

### Mongoose version

6.4.1

### Node.js version

16.15.1

### MongoDB version

5.0.9

### Operating system

Linux

### Operating system version (i.e. 20.04, 11.3, 10)

20.04

### Issue

Hi, 
I am a beginner to coding. I took a course on udemy to learn Node, express and MongoDB(mongoose).

In default MongDB find function, if I pass wrong key and value to the filter option of the find function I don't get back anything.

No documents are returned from the database. 
![image](https://user-images.githubusercontent.com/94754141/176256031-16d3cf64-3bb7-4767-9660-69816c7c57c8.png)

**When I tried to use mongoose to perform the same action on v6.4.1, I got all the records from the collection.**

![image](https://user-images.githubusercontent.com/94754141/176256751-a0c16fae-aab6-4170-818e-b3238e4e74c6.png)

Then I rolled back to v5.13.14 of mongoose and ran the same code **I got 0 record from the collection.**

![image](https://user-images.githubusercontent.com/94754141/176255070-3bbcb26c-5f5a-470b-97e3-c7d126ec7252.png)

 **_The key(something) and value(everything) does not exists in any document in the collection._**

Is it the default behaviour for the find (filter) in 6.4.1? I also checked the documentation but it doesn't say anything about passing invalid value to the filter option.

Code: 
**_tourModel.js_**
```
const mongoose = require('mongoose');

const tourSchema = new mongoose.Schema({
  name: {
    type: String,
    trim: true,
    required: [true, 'A tour must have a name.'],
  },
  duration: {
    type: Number,
    required: [true, 'A tour must have a duration'],
  },
  maxGroupSize: {
    type: Number,
    required: [true, 'A tour must have a group size'],
  },
  ratingsQuantity: {
    type: Number,
    default: 4.5,
  },
  ratingsAverage: { type: Number, default: 0 },
  price: {
    type: Number,
    required: [true, 'A tour must have a price'],
  },
  priceDiscount: Number,
  summary: {
    type: String,
    trim: true,
    required: [true, 'A tour must have a description'],
  },
  description: {
    type: String,
    trim: true,
  },
  difficulty: {
    type: String,
    required: [true, 'A tour must have a difficulty'],
  },
  imageCover: {
    type: String,
    required: [true, 'A tour must have a cover image'],
  },
  images: [String],
  createdAt: {
    type: Date,
    default: Date.now(),
  },
  startDates: [Date],
});

const Tour = mongoose.model('Tour', tourSchema);

module.exports = Tour;
```

**_tourController.js_**

```const Tour = require('../models/tourModel');

exports.getAllTours = async (req, res) => {
  try {
    // const tours = await Tour.find(req.query);
    const tours = await Tour.find({
      something: 'everything',
    });
    // const tours = await Tour.find().where('difficulty').equals('easy');
    console.log(tours.length);
    console.log(tours);
    res.status(200).json({
      status: 'success',
      results: tours.length,
      data: {
        tours,
      },
    });
  } catch (error) {
    res.status(400).json({
      status: 'fail',
      message: error,
    });
  }
};

exports.createTour = async (req, res) => {
  try {
    const newTour = await Tour.create(req.body);
    res.status(201).json({
      status: 'success',
      data: {
        tour: newTour,
      },
    });
  } catch (err) {
    res.status(400).json({
      status: 'fail',
      message: err,
    });
  }
};

exports.getTour = async (req, res) => {
  try {
    const tour = await Tour.findById(req.params.id);

    res.status(200).json({
      status: 'success',
      results: tour.length,
      data: {
        tour: tour,
      },
    });
  } catch (error) {
    res.status(400).json({
      status: 'fail',
      message: error,
    });
  }
};

exports.patchTour = async (req, res) => {
  try {
    const tour = await Tour.findByIdAndUpdate(req.params.id, req.body, {
      new: true,
      runValidators: true,
    });

    res.status(200).json({
      status: 'success',
      data: {
        tour,
      },
    });
  } catch (error) {
    res.status(400).json({
      status: 'fail',
      message: error,
    });
  }
};

exports.deleteTour = async (req, res) => {
  try {
    await Tour.findByIdAndDelete(req.params.id);
    // console.log(tour);
    res.status(204).json({
      status: 'success',
      data: null,
    });
  } catch (error) {
    res.status(400).json({
      status: 'fail',
      message: error,
    });
  }
};
```

**Comment by sean-daley** [comment]
Probably the easiest thing to do is to do a search on "strictquery" in the issues. Some of them will explain what's going on here. Here is probably one of the better ones to look at though as it's the issue opened to try and change the default behavior for 7.0. It also references a few others which might have a better explanation of what's going on.

https://github.com/Automattic/mongoose/issues/11861

In summary, this is intended behavior. You can change the behavior by setting "strictQuery" to false and it sounds like the thought is to maybe make that the default in a future major release of mongoose.

**Comment by satnamDev-Ops** [comment]
Thanks! @sean-daley
